src/index.js

```diff
@@ -0,0 +1,814 @@
+import { version } from '../package.json';
+import utils from './utils/utils';
+import requestUtils from './utils/requestUtils';
+import specUtils from './utils/specUtils';
+import specTemplates from './charts/specs';
+import * as d3 from 'd3';
+import * as vg from 'vega';
+
+export default class Cedar {
+  /**
+   * Creates a new Chart object.
+   *
+   * @example
+   *  var chart = new Cedar({
+   *    "type": "bar"
+   *    "dataset":
+   *      "url":"http://maps2.dcgis.dc.gov/dcgis/rest/services/DCGIS_DATA/Education_WebMercator/MapServer/5",
+   *      "query": {
+   *        "groupByFieldsForStatistics": "FACUSE",
+   *        "outStatistics": [{
+   *          "statisticType": "sum",
+   *          "onStatisticField": "TOTAL_STUD",
+   *          "outStatisticFieldName": "TOTAL_STUD_SUM"
+   *        }]
+   *      },
+   *      "mappings":{
+   *        "sort": "TOTAL_STUD_SUM DESC",
+   *        "x": {"field":"FACUSE","label":"Facility Use"},
+   *        "y": {"field":"TOTAL_STUD_SUM","label":"Total Students"}
+   *      }
+   *    }
+   *  });
+   *
+   * @param {Object} options
+   * @param {String} options.type - Chart type as a chartType ("bar") or a URL to a Cedar specification
+   * @param {Object} options.dataset - Dataset definition including Source and Style mappings
+   * @param {String} options.dataset.url - GeoService Layer URL
+   *
+   * "url":"http://.../rest/services/DATA/Education/MapServer/5"
+   * @param {Object} options.dataset.query - GeoServices Layer query parameters (where, bbox, outStatistics) [optional]
+   *
+   * "query": {
+   *   "groupByFieldsForStatistics": "FACUSE",
+   *   "outStatistics": [{
+   *     "statisticType": "sum",
+   *     "onStatisticField": "TOTAL_STUD",
+   *     "outStatisticFieldName": "TOTAL_STUD_SUM" }] }
+   * @param {Object} options.dataset.data - Inline feature collection, alternative to data from a URL
+   *
+   * "data": {"features":[{"attributes":{"ZIP_CODE":20005,"TOTAL_STUD_SUM":327}}]}
+   * @param {Object} options.dataset.mappings - Relates data items to the chart style definition
+   * @param {Object} options.override - Changes to the "options.type" chart specification
+   * @param {Object} options.tooltip - Optional on-hover tooltip. Element has class="cedar-tooltip" for styling.
+   * @param {String} options.tooltip.id - Optional HTML element to use for tooltip. (default: unique id created)
+   * @param {String} options.tooltip.title - Templated tooltip heading. Uses "{Variable} template format"
+   * @param {String} options.tooltip.content - Templated tooltip body text. Uses "{Variable} template format"
+   * @return {Object} new Cedar chart object
+   */
+  constructor(options) {
+    this.version = version;
+    // Pull templates in
+    this.chartTypes = specTemplates;
```

Comment:
NOTE: in v1, I expect that the global we expose will be a stateless namespace (i.e. `cdr`) and the chart types will live there instead of on the instance like you've done here, but for now, this should work.